catalog/ui/catalog-ui-search/src/main/webapp/js/model/List.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+var Backbone = require('backbone');
+var Query = require('./Query');
+var cql = require('js/cql');
+var Common = require('js/Common');
+var _ = require('lodash')
+require('backboneassociations');
+
+var iconMap = {
+    folder: 'fa fa-folder',
+    target: 'fa fa-bullseye',
+    video: 'fa fa-file-video-o',
+    text: 'fa fa-file-text-o',
+    word: 'fa fa-file-word-o',
+    powerpoint: 'fa fa-file-powerpoint-o',
+    excel: 'fa fa-file-excel-o',
+    pdf: 'fa fa-file-pdf-o',
+    image: 'fa fa-file-image-o',
+    audio: 'fa fa-file-audio-o',
+    code: 'fa fa-file-code-o',
+    archive: 'fa fa-file-archive-o',
+    tasks: 'fa fa-tasks'
+};
+
+function getRelevantIcon(iconName) {
+    return iconMap[iconName];
+}
+
+function generateCql(bookmarks) {
+    if (bookmarks.length === 0) {
+        return '';
+    }
+    return cql.write({
```

Comment:
Is there a way to do a geospatial search for a list? I'm thinking about the target list. If it contains several metacards, I would want to do a search with those location fields OR'd together.

Reply:
What if instead of the javascript building the CQL, you called into the Java code, which could then see if any CqlBuilder instances are registered for the list type?

Reply:
I think that'd be a great idea for extending list interactions in the future.  Then we could have a lot of different options for doing searches based off a list.